src/metrics.rs

```diff
@@ -39,7 +39,7 @@ fn wait_for_req_or_eof(dest: &net::TcpStream, logger: slog::Logger) -> Result<()
                 logger,
                 "failure to read request {:?}, unable to serve metrics", e
             );
-            dest.shutdown(net::Shutdown::Both);
+            let _ = dest.shutdown(net::Shutdown::Both);
```

Comment:
What is the type here? If this is a `Result` then we should consider handling it explicitly. (Here and below.)